scipy/stats/_stats_py.py

```diff
@@ -4771,16 +4813,16 @@ def statistic(x, y):
     # dtype is the data type for the calculations.  This expression ensures
     # that the data type is at least 64 bit floating point.  It might have
     # more precision if the input is, for example, np.longdouble.
-    dtype = type(1.0 + x[0] + y[0])
+    dtype = type(1.0 + x.ravel()[0] + y.ravel()[0])
 
     if n == 2:
-        r = dtype(np.sign(x[1] - x[0])*np.sign(y[1] - y[0]))
-        result = PearsonRResult(statistic=r, pvalue=1.0, n=n,
-                                alternative=alternative, x=x, y=y)
+        r = dtype(np.sign(x[..., 1] - x[..., 0])*np.sign(y[..., 1] - y[..., 0]))
```

Comment:
`axis=axis` elsewhere may be misleading, because really we rely on `axis=-1`. (I suppose we could `take_along_axis` if desired, but it's simpler to just move the axis, and might be faster to work along last axis if we take the time to copy.)

Reply:
Also, `dtype(...)` seems like a weird way to initialize the array. Why not `(...).astype(dtype)` or `np.asarray(..., dtype=dtype)`